src/main.rs

```diff
@@ -135,52 +183,59 @@ fn main() {
         Default::default()
     };
 
-    if let Some(source) = matches.opt_str("s") {
-        config.source = source;
-    };
+    config.source = matches.value_of("source")
+                           .map(str::to_string)
+                           .unwrap_or(config.source);
 
-    if let Some(dest) = matches.opt_str("d") {
-        config.dest = dest;
-    };
+    config.dest = matches.value_of("destination")
+                         .map(str::to_string)
+                         .unwrap_or(config.dest);
 
-    if let Some(layouts) = matches.opt_str("layouts") {
-        config.layouts = layouts;
-    };
+    config.layouts = matches.value_of("layouts")
+                         .map(str::to_string)
+                         .unwrap_or(config.layouts);
 
-    if let Some(posts) = matches.opt_str("posts") {
-        config.posts = posts;
-    };
+    config.posts = matches.value_of("posts")
+                         .map(str::to_string)
+                         .unwrap_or(config.posts);
 
-    let command = if !matches.free.is_empty() {
-        matches.free[0].clone()
-    } else {
-        print_usage(opts);
-        return;
-    };
+    match matches.subcommand() {
+        ("new", Some(submatches)) => {
+            match submatches.value_of("DIRECTORY") {
+                Some(directory) => config.dest = directory.to_string(),
+                _ => ()
+            }
 
-    // Check for port and set port variable to it
-    let port = matches.opt_str("port").unwrap_or("3000".to_owned());
-    let branch = matches.opt_str("branch").unwrap_or("gh-pages".to_owned());
-    let message = matches.opt_str("message").unwrap_or("cobalt site import".to_owned());
-    let should_import = matches.opt_present("import");
+            match create_new_project(&config.dest) {
+                Ok(_) => info!("Created new project at {}", config.dest),
+                Err(e) => {
+                    error!("{}", e);
+                    error!("Could not create a new cobalt project");
+                    std::process::exit(1);
+                }
+            }
+        },
 
-    match command.as_ref() {
-        "build" => {
+        ("build", Some(submatches)) => {
             build(&config);
-            if should_import {
+            if submatches.is_present("import") {
+                let branch = submatches.value_of("branch").unwrap().to_string();
```

Comment:
Okay then we can keep it. If it were to panic people can report it as an issue which is what it should be.